src/main.ts

```diff
@@ -0,0 +1,258 @@
+    repositoryReference: templateRepositoryName,
+    workingDirectoryPath,
+    cachedRepositoriesDirectoryPath,
+    outputLogger,
+  });
+
+  await lintProjects({
+    projectReferences,
+    template,
+    workingDirectoryPath,
+    cachedRepositoriesDirectoryPath,
+    outputLogger,
+  });
+}
+
+/**
+ * Parses command-line arguments and establishes parameters for the remainder of
+ * this tool.
+ *
+ * @param args - The arguments to this function.
+ * @param args.argv - The name of this executable and its arguments (as obtained
+ * via `process.argv`).
+ * @param args.outputLogger - Writable streams for output messages.
+ * @param args.defaultProjectNames - The set of MetaMask repositories
+ * that will be linted.
+ * @returns The parsed command-line arguments if parsing was successful, or null
+ * otherwise.
+ */
+async function parseInputs({
+  argv,
+  outputLogger,
+  defaultProjectNames,
+}: {
+  argv: string[];
+  outputLogger: OutputLogger;
+  defaultProjectNames: string[];
+}): Promise<Inputs | null> {
+  const args = await parseCommandLineArguments({ argv, outputLogger });
+
+  if (args) {
+    const projectReferences = args._.length > 0 ? args._ : defaultProjectNames;
+    return {
+      templateRepositoryName: DEFAULT_TEMPLATE_REPOSITORY_NAME,
+      projectReferences,
+    };
+  }
+
+  /* istanbul ignore next: At the moment, there is no real way that Yargs could fail */
+  return null;
+}
+
+/**
+ * Parses the command line using `yargs`.
+ *
+ * @param args - The arguments to this function.
+ * @param args.argv - The name of this executable and its arguments (as obtained
+ * via `process.argv`).
+ * @param args.outputLogger - Writable streams for output messages.
+ */
+async function parseCommandLineArguments({
+  argv,
+  outputLogger,
+}: {
+  argv: string[];
+  outputLogger: OutputLogger;
+}) {
+  let yargsFailure: { message: string; error: Error } | null = null;
+  const onFail = (message: string, error: Error) => {
+    /* istanbul ignore next: Not sure how to produce this */
+    if (error) {
+      throw error;
+    }
+    yargsFailure = { message, error };
+  };
+
+  const parser = yargs(hideBin(argv))
+    .usage(USAGE_TEXT)
+    .help(false)
+    .version(false)
+    .option('help', {
+      alias: 'h',
+      describe: 'Show help',
+      type: 'boolean',
+    })
+    .wrap(null)
+    .exitProcess(false)
+    .fail(onFail)
+    .strictOptions();
+
+  const options = await parser.parse();
+
+  if (options.help) {
+    outputLogger.logToStderr(await parser.getHelp());
+    return null;
+  }
+
+  if (yargsFailure) {
+    const { message } = yargsFailure;
+    outputLogger.logToStderr(
+      'ERROR: %s\n\n%s',
+      message,
+      await parser.getHelp(),
+    );
+    return null;
+  }
+
+  return {
+    ...options,
+    // This is the name that Yargs gives to the arguments that aren't options.
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    _: options._.map(String),
```

Comment:
Nit: Perhaps this is an opportunity to give this a better name?

Reply:
Good point! Updated here: [ca73b38](https://github.com/MetaMask/module-lint/pull/33/commits/ca73b387df05edbead39a25364876906b10e83d4)